tests/unit/database_tests.py

```diff
@@ -1065,5 +1066,108 @@ def test_get_query_indexes(self):
         self.assertEqual(indexes[2].design_document_id, '_design/ddoc001')
         self.assertEqual(indexes[2].name, 'text-idx-001')
 
+    def test_get_search_result_with_invalid_argument(self):
+        """
+        Test get_search_result by passing in invalid arguments
+        """
+        try:
+            self.db.get_search_result('searchddoc001', 'searchindex001',
+                                      'julia*', foo={'bar': 'baz'})
+            self.fail('Above statement should raise an Exception')
+        except CloudantArgumentError as err:
+            self.assertEqual(str(err), 'Invalid argument: foo')
```

Comment:
Rather than using the try/except block to test invalid conditions you should use `assertRaises`.  See this #80 for detail.  This applies to all new tests that you added as part of this PR.